src/charts/axis.js

```diff
@@ -196,6 +197,19 @@ export default function Axis (_container) {
     }
   }
 
+  function truncateTickText (axisSelection) {
+    if (!config.maxLabelCharCount) {
```

Comment:
A positive condition without an early return here (just `if (config.maxLabelCharCount) { axis... }`) would be a little more straightforward, airtight, and easier to extend / change in the future.